app/src/main/scala/org/alephium/explorer/persistence/queries/HashrateQueries.scala

```diff
@@ -33,33 +34,42 @@ trait HashrateQueries extends CustomTypes {
   def getHashratesQuery(from: TimeStamp,
                         to: TimeStamp,
                         interval: Int): DBActionSR[(TimeStamp, BigDecimal)] = {
-    val fromInstant = Instant.ofEpochMilli(from.millis)
-    val toInstant   = Instant.ofEpochMilli(to.millis)
 
     sql"""
         SELECT timestamp, value
         FROM hashrates
-        WHERE interval_type = #$interval
-        AND timestamp >= '#${fromInstant.toString}'
-        AND timestamp <= '#${toInstant.toString}'
+        WHERE interval_type = $interval
+        AND timestamp >= $from
+        AND timestamp <= $to
         ORDER BY timestamp
       """.as[(TimeStamp, BigDecimal)]
   }
 
   def computeHashratesAndInsert(from: TimeStamp, intervalType: Int): DBActionW[Int] = {
-    val hashrates = (intervalType match {
-      case 0 => compute10MinutesHashrateRawString(from)
-      case 1 => computeHourlyHashrateRawString(from)
-      case 2 => computeDailyHashrateRawString(from)
-      case _ => "()"
-    })
-
-    sqlu"""
-      INSERT INTO hashrates (timestamp, value, interval_type)
-      (#$hashrates)
-      ON CONFLICT (timestamp, interval_type) DO UPDATE
-      SET value = EXCLUDED.value
-    """
+    if (intervalType >= 0 || intervalType <= 2) {
+      val dateGroup = (intervalType match {
+        case 0 => tenMinutesQuery
+        case 1 => hourlyQuery
+        case 2 => dailyQuery
+        case _ => ""
+      })
+
+      sqlu"""
+        INSERT INTO hashrates (timestamp, value, interval_type)
```

Comment:
Shall we store the hashrates as one table or several separate tables?

Reply:
As we have an index on the interval_type, I'm not sure it's worth it.

Reply:
Since we created index for `interval_type`, it is fine